src/preferencesControl.ts

```diff
@@ -217,6 +217,85 @@ export class PreferencesControl {
+        let modalButtonReset: HTMLInputElement = <HTMLInputElement> document.getElementsByClassName(styles.modalButtonReset)[0];
+
+        let acceptRejectButtons: Element[] = [].slice.call(document.getElementsByClassName(styles.cookieItemRadioBtn));
+
+        let lastAcceptRadioBtn: HTMLElement | null = null;
+        let lastRejectRadioBtn: HTMLElement | null = null;
+        if (acceptRejectButtons.length) {
+            lastAcceptRadioBtn = <HTMLElement> acceptRejectButtons[acceptRejectButtons.length - 2];
+            lastRejectRadioBtn = <HTMLElement> acceptRejectButtons[acceptRejectButtons.length - 1];
+        }
+
+        let lastElementTab = function(event: KeyboardEvent): void {
+            if (event.key == 'Tab' && !event.shiftKey) {
+                event.preventDefault();
+                closeModalIcon.focus();
+            }
+        };
+        let firstElementShiftTab = function(event: KeyboardEvent): void {
+            if (event.key == 'Tab' && event.shiftKey) {
+                event.preventDefault();
+                lastRejectRadioBtn?.focus();
+            }
+        };
+
+        modalButtonReset?.addEventListener('keydown', (event) => {
+            if (event.key == 'Tab' && !event.shiftKey) {
+                event.preventDefault();
+                closeModalIcon.focus();
+            }
+        });
+
+        if (modalButtonReset.disabled && modalButtonSave.disabled) {
+            if (acceptRejectButtons.length) {
+                lastAcceptRadioBtn?.addEventListener('keydown', lastElementTab);
+                lastRejectRadioBtn?.addEventListener('keydown', lastElementTab);
+
+                closeModalIcon?.addEventListener('keydown', firstElementShiftTab);
+            }
+        }
+        else {
+            closeModalIcon?.addEventListener('keydown', (event) => {
+                if (event.key == 'Tab' && event.shiftKey) {
+                    event.preventDefault();
+                    modalButtonReset.focus();
+                }
+            });
+        }
+
+        if (acceptRejectButtons.length) {
+            for (let radio of acceptRejectButtons) {
+                radio.addEventListener('click', () => {
+                    // Enable "Reset all" button
+                    // Update event listener function in "X" and remove event listener in last accept/reject radio buttons
+                    if (modalButtonReset.disabled) {
+                        modalButtonReset.disabled = false;
+
+                        lastAcceptRadioBtn?.removeEventListener('keydown', lastElementTab);
+                        lastRejectRadioBtn?.removeEventListener('keydown', lastElementTab);
+                        
+                        closeModalIcon?.removeEventListener('keydown', firstElementShiftTab);
+                        closeModalIcon?.addEventListener('keydown', (event) => {
+                            if (event.key == 'Tab' && event.shiftKey) {
```

Comment:
This function looks the same as on line 269.
Please extract it to a separate function.